app/controllers/posts_controller.rb

```diff
@@ -1,13 +1,14 @@
 # Post Controller
 class PostsController < ApplicationController
-  before_action :find_post, only: %i[show edit update destroy]
+  before_action :find_post, only: %i[show edit update destroy add_donator]
   before_action :authenticate_user!, except: %i[index show]
 
   def index
     @posts = Post.all
   end
 
   def specific_indexing
```

Comment:
This function is used to show only those posts to users that match their blood-type. For example, a user with BloodType B+ will only see posts that require B+ blood-type. 

I have added this so that only Admins can see all the posts where as normal users will only see posts related to them.